### PR TITLE
CP-49078: Preprocess fields into a Hashtbl within get_record

### PR DESCRIPTION
**Pull request by contificate**
Flame graphs indicate that, under load created by parallel "xe vm-list" commands, the DB action get_record is hit often. This function constructs an API-level record by marshalling an association list that maps field names to unmarshalled string values. To do this, it serially queries all the field names using `List.assoc`. This has rather large cost in doing lexicographical string comparisons (`caml_compare` on string keys).

To avoid this, regardless of record size, we preprocess the association lists `__regular_fields` and `__set_refs` into a (string, string) Hashtbl.t and query that to construct each record field.

---

This benefit of this change is most notable for large records (such as `VM` and `pool`). The cost of the previously generated code, which does a bunch of serial `List.assoc` calls, incurs the quadratic cost of list traversal (compounded by the costly lexicographical comparison of strings during the search).

To produce measurements, I sampled xapi under a load of 500 consecutive `xe vm-list` invocations (using the same sampling rate with `perf`) on a host with a single VM.

Without the change, the `get_record` done internally by `xe vm-list` makes up for ~33.59% of the samples (33.59% = 1,592,782,255 samples). With the change, `get_record` accounts for ~7.56 of the samples (of which there are substantially fewer collected: 7.56% = 264,948,239). So the number of samples for `get_record` has dropped from 1,592,782,255 to 264,948,239 (assuming `perf`'s sampling is reliable).

You can see the visual difference in the flame graphs:

Before:
{B1FEFF3A-AD91-478B-A828-89DCD19C2BEA}

After:
{CB28FFEC-944D-4F26-918A-FFB57E4875A3}

Of course, this is benefit as measured in aggregate (`perf` sampling), so quite a fast and loose comparison. In practice, the `xe vm-list` stress test goes from 7.4s to 6.2s (as `get_record` makes up only a small part of the work done for a single `xe vm-list`).

### CODE REVIEW

File: ocaml/idl/datamodel_lifecycle.ml [hunk 1]
```diff
@@ -52,7 +52,7 @@ let prototyped_of_field = function
   | "VTPM", "persistence_backend" ->
       Some "22.26.0"
   | "SM", "host_pending_features" ->
-      Some "24.36.0-next"
+      Some "24.37.0"
```
Comment (psafont) [inline ocaml/idl/datamodel_lifecycle.ml:55]: This should be an independent commit

**Review by psafont — APPROVED** [review]

File: ocaml/database/schema.ml [hunk 2]
```diff
@@ -96,28 +96,104 @@ module Column = struct
           (** only so we can special case set refs in the interface *)
   }
   [@@deriving sexp]
+
+  let name_of t = t.name
 end
 
+let tabulate ks ~key_fn =
+  let tbl = Hashtbl.create 64 in
+  List.iter (fun c -> Hashtbl.add tbl (key_fn c) c) ks ;
```
Comment (lindig) [inline ocaml/database/schema.ml:105]: `Hashtbl.replace`?

**Review by lindig — APPROVED** [review]
How are tables shrining or removed?
